shared/middleware/net_sim_api.rb

```diff
@@ -0,0 +1,111 @@
+require 'sinatra/base'
+require 'cdo/db'
+require 'cdo/rack/request'
+require 'csv'
+
+class NetSimApi < Sinatra::Base
+
+  helpers do
+    [
+        'core.rb',
+        'storage_id.rb',
+        'table.rb',
+    ].each do |file|
+      load(CDO.dir('shared', 'middleware', 'helpers', file))
+    end
+  end
+
+  TableType = CDO.use_dynamo_tables ? DynamoTable : Table
+
+  # Environment-dependent channel ID for NetSim.
+  # Will no longer be needed when we detach from the channels API entirely.
+  def netsim_channel_id
+    rack_env?(:development) ? 'JGW2rHUp_UCMW_fQmRf6iQ==' : 'HQJ8GCCMGP7Yh8MrtDusIA=='
+  end
+
+  def channels_table_name(shard_id, table_name)
+    "#{shard_id}_#{table_name}"
+  end
+
+  #
+  # GET /v3/netsim/<shard-id>/<table-name>
+  #
+  # Returns all of the rows in the table.
+  #
+  get %r{/v3/netsim/([^/]+)/(\w+)$} do |shard_id, table_name|
+    dont_cache
+    content_type :json
+    TableType.new(netsim_channel_id, storage_id('shared'), channels_table_name(shard_id, table_name)).to_a.to_json
```

Comment:
To reduce duplicated code, please create a helper method for getting the table given the shard_id and table_name and use it through this class.